buildSrc/build.gradle.kts

```diff
@@ -13,25 +13,89 @@
  * limitations under the License.
  */
 
+import org.jetbrains.kotlin.gradle.tasks.KotlinCompile
+
 plugins {
     `kotlin-dsl`
 }
 
 repositories {
+    mavenCentral()
     gradlePluginPortal()
     google()
-    mavenCentral()
 }
 
 dependencies {
-    implementation("org.jetbrains.kotlin:kotlin-gradle-plugin:1.6.21")
-    implementation("com.android.tools.build:gradle:4.1.0")
-
-    implementation("com.diffplug.spotless:spotless-plugin-gradle:6.5.2")
-    implementation("com.google.protobuf:protobuf-gradle-plugin:0.9.1")
-    implementation("io.codearte.gradle.nexus:gradle-nexus-staging-plugin:0.30.0")
-    implementation("com.google.guava:guava:31.1-jre")
-    implementation("ru.vyarus:gradle-animalsniffer-plugin:1.5.4")
-    implementation("org.jetbrains.kotlinx:binary-compatibility-validator:0.8.0")
+    implementation(libs.androidGradlePlugin)
+    implementation(libs.animalSnifferGradlePlugin)
+    implementation(libs.binaryCompatibilityValidator)
+    implementation(libs.gradleMavenPublishPlugin)
+    implementation(libs.kotlinGradlePlugin)
+    implementation(libs.protobuf.gradlePlugin)
+    implementation(libs.spotlessGradlePlugin)
     implementation(kotlin("gradle-plugin-api"))
 }
+
+tasks.withType<KotlinCompile> {
+    dependsOn("generateVersions")
+}
+
+val versionOutputDir = layout.buildDirectory.file("generated/protobuf-version")
+
+sourceSets["main"].java.srcDir(versionOutputDir)
+
+tasks.register("generateVersions") {
+    doFirst {
```

Comment:
make these into proper tasks with inputs/outputs declared

alternatively, use the build config plugin (see https://github.com/open-toast/can-i-use-anvil/blob/main/can-i-use-anvil-plugin/build.gradle.kts#L10 for an example)

Reply:
Done.